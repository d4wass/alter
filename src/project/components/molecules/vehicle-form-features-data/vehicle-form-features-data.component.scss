:host {
  display: flex;
  justify-content: center;

  form {
    width: 60%;
  }

  .mat-radio-button,
  .mat-checkbox {
    margin-right: 20px;
  }
}
