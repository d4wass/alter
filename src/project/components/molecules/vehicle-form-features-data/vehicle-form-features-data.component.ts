import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'app-vehicle-form-features-data',
  template: `
    <form>
      <h2>Features</h2>
      <h4>Engine</h4>
      <mat-form-field appearance="fill" floatLabel="always">
        <mat-label>capacity</mat-label>
        <input matInput type="number" class="example-right-align" placeholder="0" />
      </mat-form-field>
      <mat-form-field appearance="fill" floatLabel="always">
        <mat-label>power</mat-label>
        <input matInput type="number" class="example-right-align" placeholder="0" />
      </mat-form-field>
      <h4>Gearbox</h4>
      <mat-radio-group aria-label="Select an option">
        <mat-radio-button value="manual">Manual</mat-radio-button>
        <mat-radio-button value="automatic">Automatic</mat-radio-button>
      </mat-radio-group>
      <h4>Drive type</h4>
      <mat-radio-group aria-label="Select an option">
        <mat-radio-button value="rear">rear</mat-radio-button>
        <mat-radio-button value="front">front</mat-radio-button>
        <mat-radio-button value="all">all</mat-radio-button>
      </mat-radio-group>
      <h4>Additional Equipment</h4>
      <section class="example-section">
        <mat-checkbox class="example-margin">Apple Car Play</mat-checkbox>
        <mat-checkbox class="example-margin">Android Auto</mat-checkbox>
        <mat-checkbox class="example-margin">Bluetooth</mat-checkbox>
        <mat-checkbox class="example-margin">USB</mat-checkbox>
        <mat-checkbox class="example-margin">headup display</mat-checkbox>
        <mat-checkbox class="example-margin">navigation</mat-checkbox>
        <mat-checkbox class="example-margin">air conditioning</mat-checkbox>
        <mat-checkbox class="example-margin">keyless access</mat-checkbox>
        <mat-checkbox class="example-margin">isofix</mat-checkbox>
        <section>
          <h5>Cruise Control</h5>
          <mat-checkbox class="example-margin">standard</mat-checkbox>
          <mat-checkbox class="example-margin">active</mat-checkbox>
          <mat-checkbox class="example-margin">adaptive</mat-checkbox>
          <mat-checkbox class="example-margin">autonomic</mat-checkbox>
        </section>
        <section>
          <h5>Lights</h5>
          <mat-checkbox class="example-margin">led</mat-checkbox>
          <mat-checkbox class="example-margin">xenon</mat-checkbox>
          <mat-checkbox class="example-margin">bi-xenon</mat-checkbox>
          <mat-checkbox class="example-margin">laser</mat-checkbox>
        </section>
        <section>
          <h5>Park assist</h5>
          <mat-checkbox class="example-margin">camera</mat-checkbox>
          <mat-checkbox class="example-margin">camera 360</mat-checkbox>
          <mat-checkbox class="example-margin">automatic</mat-checkbox>
        </section>
      </section>
    </form>
  `,
  styleUrls: ['./vehicle-form-features-data.component.scss']
})
export class VehicleFormFeaturesDataComponent {
  constructor() {}
}
