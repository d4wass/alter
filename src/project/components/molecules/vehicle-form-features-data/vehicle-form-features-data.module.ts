import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { VehicleFormFeaturesDataComponent } from './vehicle-form-features-data.component';
import { MatFormFieldModule } from '@angular/material/form-field';
import { MatCheckboxModule } from '@angular/material/checkbox';
import { MatRadioModule } from '@angular/material/radio';
import { MatInputModule } from '@angular/material/input';
import { ReactiveFormsModule } from '@angular/forms';
import { VehicleFeatureCheckboxInputFormModule } from '../../atoms/vehicle-feature-checkbox-input-form/vehicle-feature-checkbox-input-form.module';
import { VehicleFeatureRadioInputFormModule } from '../../atoms/vehicle-feature-radio-input-form/vehicle-feature-radio-input-form.module';

@NgModule({
  declarations: [VehicleFormFeaturesDataComponent],
  imports: [
    CommonModule,
    MatFormFieldModule,
    MatRadioModule,
    MatCheckboxModule,
    MatInputModule,
    ReactiveFormsModule,
    VehicleFeatureCheckboxInputFormModule,
    VehicleFeatureRadioInputFormModule
  ],
  exports: [VehicleFormFeaturesDataComponent]
})
export class VehicleFormFeaturesDataModule {}
