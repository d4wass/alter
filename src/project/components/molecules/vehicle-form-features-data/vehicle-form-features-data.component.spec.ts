import { ComponentFixture, TestBed } from '@angular/core/testing';

import { VehicleFormFeaturesDataComponent } from './vehicle-form-features-data.component';

describe('VehicleFormFeaturesDataComponent', () => {
  let component: VehicleFormFeaturesDataComponent;
  let fixture: ComponentFixture<VehicleFormFeaturesDataComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [VehicleFormFeaturesDataComponent]
    }).compileComponents();

    fixture = TestBed.createComponent(VehicleFormFeaturesDataComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
