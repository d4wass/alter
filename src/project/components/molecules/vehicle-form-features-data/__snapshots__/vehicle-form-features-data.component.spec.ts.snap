// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleFormFeaturesDataComponent should match to snapshot 1`] = `
<app-vehicle-form-features-data
  driveCtrl={[Function FormControl]}
  driveTypes={[Function Array]}
  equipmentCtrl={[Function FormControl]}
  formGroupCtrl={[Function FormGroup]}
  gearboxCtrl={[Function FormControl]}
  gearboxTypes={[Function Array]}
>
  <form
    class="ng-untouched ng-pristine ng-valid"
    novalidate=""
  >
    <h2>
      Features
    </h2>
    <form
      class="ng-untouched ng-pristine ng-valid"
      formgroupname="engine"
      novalidate=""
    >
      <h4>
        Engine
      </h4>
      <mat-form-field
        appearance="fill"
        floatlabel="always"
      >
        <mat-label>
          capacity
        </mat-label>
        <input
          class="example-right-align ng-untouched ng-pristine ng-valid"
          formcontrolname="capacity"
          matinput=""
          placeholder="0"
          type="number"
        />
      </mat-form-field>
      <mat-form-field
        appearance="fill"
        floatlabel="always"
      >
        <mat-label>
          power
        </mat-label>
        <input
          class="example-right-align ng-untouched ng-pristine ng-valid"
          formcontrolname="power"
          matinput=""
          placeholder="0"
          type="number"
        />
      </mat-form-field>
    </form>
    <app-vehicle-feature-radio-input-form
      title="Gearbox"
    />
    <app-vehicle-feature-radio-input-form
      title="Drive"
    />
    <h4>
      Additional Equipment
    </h4>
    <app-vehicle-equipment-input-form />
  </form>
</app-vehicle-form-features-data>
`;
